core/server/worker/src/main/java/alluxio/worker/grpc/GrpcDataServer.java

```diff
@@ -99,7 +99,7 @@ public GrpcDataServer(final String hostName, final SocketAddress bindAddress,
               workerProcess, mDomainSocketAddress != null);
       mServer = createServerBuilder(hostName, bindAddress, NettyUtils.getWorkerChannel(
           Configuration.global()))
-          .addService(ServiceType.FILE_SYSTEM_WORKER_WORKER_SERVICE, new GrpcService(
+          .addService(ServiceType.BLOCK_WORKER_CLIENT_SERVICE, new GrpcService(
```

Comment:
Instead of change the service type here, should we add the service twice, so that the service can be found with the two different service names, for the sake of backwards compatibility? Did you test with an older client with a new worker after this change?

Reply:
In practice old worker clients don't check version, so this should be fine.

Reply:
Other than version check, does not the client use the service type in any way when communicating with the worker? e.g. for other RPC method calls?

Reply:
Nope. This type is used in `WorkerHealthCheckClient` only, which runs inside alluxio servers. 

But it came up to me that this change does break "forward compatibility": after worker client is changed, it won't work with older versions of alluxio server. Would that be a concern for us?

Reply:
@jiacheliu3 how do we handle gRPC definition breaking changes? a new client may not be able to connect to an old worker after this change.

Reply:
There might be scenarios where a new client will connect to an old worker. For example an OS user keeps up with our `master` or edge released versions and performs a rolling restart in the cluster. And during the restart there's no guarantee the client comes up after all workers (for example Presto is a client). 

However, it depends on us whether we want to support that case. After all there will be breaking changes to RPC like we add a new RPC, it is impossible to make old master/worker answer new RPCs gracefully. I think it's unavoidable that one end will break during a rolling upgrade and should automatically fix itself when the other side is ready. 

Also, AFAIK it's more challenging to upgrade the client side in practice, because our clients are everywhere baked into application processes, services and scripts etc, and they are controlled by various platform/app users. Upgrading the cluster side is much easier because it is handled by the admin. So generally speaking we expect the new cluster to handle old clients gracefully.

Reply:
To sum up, this particular PR and its downstream #15837 will have the following impact on backward compatibility:

## Before
- An old worker has a version service under the name `FILE_SYSTEM_WORKER_WORKER_SERVICE`* and an unknown version number (`-1`).
- An old `BlockWorkerClient` does not check the worker's version after connecting to the worker.

## After
- A new worker has a version service under the name `BLOCK_WORKER_CLIENT_SERVICE`* and a dedicated version number (`1`).
- A new client will check the worker's version and throw an exception if the versions are not equal.

Therefore, the breaking change is when a new client connects to an old worker, an unexpected exception will make the client refuse to connect to the worker.

*: IMO, renaming a symbol in gRPC definition but keeping its enum tag unchanged should not be a breaking change, but [proto-backwards-compat-maven-plugin](https://github.com/salesforce/proto-backwards-compat-maven-plugin) thinks otherwise.

Reply:
Updated the PR description.